merlion/utils/resample.py

```diff
@@ -96,38 +98,55 @@ def granularity_str_to_seconds(granularity: Union[str, float, int, None]) -> Uni
     if isinstance(granularity, (float, int)):
         ms = np.floor(granularity * 1000)
     else:
-        try:
-            granularity = pd.Timedelta(granularity)
-        except:
-            pass
         ms = np.floor(pd.tseries.frequencies.to_offset(granularity).nanos / 1e6)
-    return ms / 1000
+    return (ms / 1000).item()
 
 
-def get_gcd_timedelta(*time_stamp_lists):
+def get_date_offset(time_stamps: pd.DatetimeIndex, reference: pd.DatetimeIndex) -> pd.DateOffset:
     """
-    Calculates all timedeltas present in any of the lists of time stamps given,
-    and returns the GCD of all these timedeltas (up to units of milliseconds).
+    Returns the date offset one must add to ``time_stamps`` so its last timestamp aligns with that of ``reference``.
     """
-    all_dt = np.concatenate([np.diff(t) for t in time_stamp_lists])
-    all_dt = np.unique((all_dt * 1000).astype(np.int64))
-    gcd_dt = all_dt[0]
-    for dt in all_dt[1:]:
-        gcd_dt = np.gcd(gcd_dt, dt)
-    return gcd_dt.astype(float) / 1000
+    df, tf = time_stamps[-1], reference[-1]
```

Comment:
Hi @aadyotb, could we also consider yearly data at this PR, as leap year has one more day.

Reply:
Hi @chenghaoliu89, handling monthly granularity implicitly handles yearly granularities. This is because we allow an arbitrary number of months to be detected, e.g. 12 or 24. A 12 month granularity is therefore able to handle leap year successfully.

Reply:
Note that there is test coverage for this behavior in `tests/transform/test_resample.py`.